lib/puppet/functions/hiera_vault.rb

```diff
@@ -27,11 +27,18 @@
   def lookup_key(key, options, context)
 
     if confine_keys = options['confine_to_keys']
-      raise ArgumentError, 'confine_to_keys must be an array' unless confine_keys.is_a?(Array)
-      confine_keys.map! { |r| Regexp.new(r) }
+      raise ArgumentError, '[hiera-vault] confine_to_keys must be an array' unless confine_keys.is_a?(Array)
+
+      begin
+        confine_keys = confine_keys.map { |r| Regexp.new(r) }
+      rescue StandardError => e
+        raise Exception, "[hiera-vault] creating regexp failed with: #{e}"
+      end
+
       regex_key_match = Regexp.union(confine_keys)
+
       unless key[regex_key_match] == key
-        context.explain { "Skipping hiera_vault backend because key does not match confine_to_keys" }
+        context.explain { "[hiera-vault] Skipping hiera_vault backend because key does not match confine_to_keys" }
```

Comment:
There is no need to tag with `[hiera-vault]` as the explain features knows where the errors are coming from and will show that to the user. There is no harm though - perhaps more information is better.